SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp

```diff
@@ -629,17 +647,21 @@ void ASpatialFunctionalTest::OnReplicated_bPreparedTest()
 {
 	if (bPreparedTest)
 	{
-		if (!HasAuthority())
-		{
-			PrepareTest();
-		}
+		// We need to delay until next Tick since on non-Authority
+		// OnReplicated_bPreparedTest() will be called before BeginPlay().
```

Comment:
Why do we need to wait until after begin play is called?

Reply:
BeginPlay atm sets some variables, so what would happen is that PrepareTest would be on non-auth workers (clients + servers) called before those variables are setup. So unless we'd change how these things are handled (e.g. done only once when PrepareTest is called instead of BeginPlay) we need to ensure that BeginPlay is called first.